packages/core/src/modules/credentials/repository/CredentialRecord.ts

```diff
@@ -26,6 +27,8 @@ export interface CredentialRecordProps {
   threadId: string
 
   credentialId?: string
+  revocationRegistryId?: string
```

Comment:
We're in the process of making the credential record non indy specific. Maybe you can extend the metadata type for the indy credential and store it alongside the schemaId and credentialDefinitionId. This will save some migrations soon

Reply:
> revocationRegistryId and credentialRevocationId need to be saved to the credential record. We are storing them as properties of CredentialRecord, I know that storing them in metadata may also be an option. If we wanted to store them as metadata then some guidance will be needed.

We were thinking the same thing ^ , but had some difficulty understanding the black magic happening here 😆 

Additionally there's a grossness in that when we can only get the cred_rev_id out of the credential in getCredential() and not when we're storing the credential, which makes it more challenging.

Reply:
Ah let me guide you. In `credentialMetadataTypes.ts` extend the  value of the `CredentialMetadataKeys.IndyCredential` key with the `revocationRegistryId` and the `credentialRevocationId`. 

You can now use it like this:

```ts
const indyCredential = credentialRecord.metadata.get(CredentialMetadataKeys.IndyCredential)

indyCredential?.revocationRegistryId

// .add adds to the existing metadata object, creating it if it doesn't exist yet
credentialRecord.metadata.add(CredentialMetadataKeys.IndyCredential, {
   revocationRegistryId: 'a rev id',
   credentialRevocationId: 'the value'
})
```

You should make sure to save the record afterwards

Reply:
On the `cred_rev_id` only be present when retrieving the credential. It seems that ACA-Py extract the `cred_rev_id` from the received credential attachment: https://github.com/hyperledger/aries-cloudagent-python/blob/main/aries_cloudagent/protocols/issue_credential/v2_0/formats/indy/handler.py#L517

Could it be that our typing is not reflecting the actual object? Otherwise I don't see a problem with storing + fetching it directly afterwards. I think receiving a credential will be done a lot less than showing it for proof, so don't see the problem in one extra DB query. However indy has to extract it from somewhere, so it should be in the received `Cred` type right?

Reply:
Awesome, thank you for the insight on the metadata bit here--we'll let you know if we run into any other struggles there. 

Also, thank you for the insight into ACA-Py's implementation, we'll re-review if we can access the cred_rev_id in the credential attachment. 👍

Reply:
Hey @TimoGlastra, I was looking at adding the revocationRegistryId and credentialRevocationId to the credential metadata instead of the credential record when I ran into an issue. We need to be able to search for a credential based off of those two properties, I am not seeing a clear way to do that if they are only in the metadata. Any insight here would be appreciated.

Reply:
Interesting. Maybe you could access the metadata from inside the `getTags` method?

This is an example with the (already existing) credentialDefinitionId. I think you could do the same for `credentialRevocationId`:

```ts
  public getTags() {
    const metadata = this.metadata.get(CredentialMetadataKeys.IndyCredential)

    return {
      ...this._tags,
      threadId: this.threadId,
      connectionId: this.connectionId,
      state: this.state,
      credentialId: this.credentialId,
      credentialDefinitionId: metadata?.credentialDefinitionId,
    }
  }
```